account_asset_management_number/models/account_asset.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountAsset(models.Model):
+    _inherit = "account.asset"
+    _rec_name = "number"
+
+    number = fields.Char(
+        string="Asset Number",
+        default="",
+        index=True,
+        copy=False,
+    )
+    number_by_sequence = fields.Boolean(related="profile_id.number_by_sequence")
+
+    @api.model
+    def create(self, vals):
+        if vals.get("number", "") == "" and vals.get("profile_id"):
+            asset_profile = self.env["account.asset.profile"].browse(vals["profile_id"])
+            if asset_profile.number_by_sequence and asset_profile.sequence_id:
+                vals["number"] = asset_profile.sequence_id.next_by_id()
+        return super().create(vals)
+
+    def _insert_number_before_name(self, list_fields):
+        i = list_fields.index("name")
+        list_fields.insert(i, "number")
+        return list_fields
+
+    @api.model
+    def _xls_acquisition_fields(self):
+        acquisition_fields = super()._xls_acquisition_fields()
+        if "name" in acquisition_fields:
+            self._insert_number_before_name(acquisition_fields)
+        return acquisition_fields
+
+    @api.model
+    def _xls_active_fields(self):
+        active_fields = super()._xls_active_fields()
+        if "name" in active_fields:
+            self._insert_number_before_name(active_fields)
+        return active_fields
+
+    @api.model
+    def _xls_removal_fields(self):
+        removal_fields = super()._xls_removal_fields()
+        if "name" in removal_fields:
+            self._insert_number_before_name(removal_fields)
```

Comment:
Why not just 1 line?
```suggestion
        list_fields.insert(list_fields.index("name"), "number")
```
`removal_fields` will always have "name", right?

Reply:
Thank you for your suggestion. I already edited it.